### DISCUSSION TITLE
Use targets for production analyses

### DISCUSSION THREAD

**Discussion by kent37** [ropensci/targets#724] [Help]
Hi, Is there a lightweight way to re-use a `{targets}` pipeline with different data sets?
What I would like is a way to define a processing pipeline that

- runs locally on Windows computers
- runs a defined sequence of processing commands
- accepts multiple input files and creates multiple intermediate and output files
- tracks dependencies and can restart a process from an intermediate step in case of failure
- is easily configurable to run on a new dataset

`{targets}` can do the first four, I'm trying to figure out if it can do the last item. I think a target factory comes close; can I use a factory without creating a new project for each target? I would love be able to edit a configuration file with source and output files / directories, then run a defined pipeline to create new outputs from the defined inputs.

Suggestions, examples, pointers all welcome, thanks!

**Comment by wlandau** [comment]
Multi-project workflows may come close to this: http://books.ropensci.org/targets/projects.html. That will give you a different data store for each version of the pipeline. Then your data target could depend on the `TAR_PROJECT` environment variable which you could set in a project-level `.Renviron` file. Sketch:

```{r}
# _targets.R file common to all projects:
library(targets)
source("R/functions.R) # defines data_path_of_project()

# Set a global variable for the data path. The data target
# depends on this path, and the dependency relationship
# is visible in tar_visnetwork().
data_path <- data_path_of_project(Sys.getenv("TAR_PROJECT"))

list(
  tar_target(data, data_path, format = "file"),
  # other targets...
)
```
